### ISSUE TITLE
 overflow detected, setting loss scale to: 1024.0 but model is sitll being trained.

### ISSUE THREAD

**Issue by Marvinmw** [pytorch/fairseq#2404]
Hi,
When I train RoBERTa, I have many the warnings like, 
2020-08-01 12:39:16 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 1024.0
2020-08-01 12:39:39 | INFO | train_inner | epoch 001:   3615 / 6547 loss=0.439, nll_loss=0.001, accuracy=87.9, wps=82565, ups=2.66, wpb=31075.5, bsz=64, num_updates=3600, lr=9e-06, gnorm=3.952, loss_scale=1399, train_wall=38, wall=1353

Is it fine?
My command is 
WARMUP_UPDATES=4000      # 6 percent of the number of updates
LR=1e-05                # Peak LR for polynomial LR scheduler.
HEAD_NAME=fariseq_head_pre     # Custom name for the classification head.
NUM_CLASSES=2           # Number of classes for the classification task.
MAX_SENTENCES=8         # Batch size.
ROBERTA_PATH=fairseq/roberta.base/model.pt
SAVE_DIR=roberta_class_base_pre
fairseq-train fairseq/fairseq_data-bin \
    --restore-file $ROBERTA_PATH \
    --max-positions 512 \
    --max-sentences $MAX_SENTENCES \
    --max-tokens 4400 \
    --task sentence_prediction \
    --reset-optimizer --reset-dataloader --reset-meters \
    --required-batch-size-multiple 1 \
    --init-token 0 --separator-token 2 \
    --arch roberta_base \
    --criterion sentence_prediction \
    --classification-head-name $HEAD_NAME \
    --num-classes $NUM_CLASSES \
    --dropout 0.1 --attention-dropout 0.1 \
    --weight-decay 0.1 --optimizer adam --adam-betas "(0.9, 0.98)" --adam-eps 1e-06 \
    --clip-norm 0.0 \
    --lr-scheduler polynomial_decay --lr $LR --warmup-updates $WARMUP_UPDATES   \
    --fp16 --fp16-init-scale 4 --threshold-loss-scale 1 --fp16-scale-window 128 \
    --max-epoch 40 \
    --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric \
    --shorten-method "truncate" \
    --find-unused-parameters \
    --update-freq 4 \
    --save-dir $SAVE_DIR

**Comment by myleott** [comment]
Yes, that's expected when training with mixed precision (`--fp16`). It's just a notice, you can ignore it as long as you don't see the scale get too small... any scale >= 0.1 is probably fine.

**Comment by edosyhptra** [comment]
> Yes, that's expected when training with mixed precision (`--fp16`). It's just a notice, you can ignore it as long as you don't see the scale get too small... any scale >= 0.1 is probably fine.

Thanks for the answer. I got the same warning in training. However, What is actually happening when it detects an overflow?
